code/game/machinery/launch_pad.dm

```diff
@@ -349,94 +358,36 @@
 	if(..())
 		return
 	switch(action)
-		if("right")
-			if(!pad.teleporting)
-				if(pad.x_offset < pad.range)
-					pad.x_offset++
-					pad.update_indicator()
+		if("set_pos")
+			var/new_x = text2num(params["x"])
+			var/new_y = text2num(params["y"])
+			if(!isnull(new_x))
+				pad.set_x(new_x)
+			if(!isnull(new_y))
+				pad.set_y(new_y)
 			. = TRUE
```

Comment:
You need to call pad.update_indicator()

Reply:
I noticed the indicator wasn't working but just assumed it was broken due to other things lmao